pkg/controller/argocd/util.go

```diff
@@ -701,10 +701,15 @@ func watchResources(c controller.Controller, clusterRoleBindingMapper handler.To
 	handlerClusterRoleBinding := &handler.EnqueueRequestsFromMapFunc{
 		ToRequests: clusterRoleBindingMapper,
 	}
+
 	if err := c.Watch(&source.Kind{Type: &v1.ClusterRoleBinding{}}, handlerClusterRoleBinding); err != nil {
 		return err
 	}
 
+	if err := c.Watch(&source.Kind{Type: &v1.ClusterRole{}}, handlerClusterRoleBinding); err != nil {
```

Comment:
Even though these cluster roles are common for all argocd instances the controller will have to watch these resources.  Here I've used the existing mapper function.

Reply:
Clusterroles should be different for each instance.